src/components/ObsDetails/DQAContainer.js

```diff
@@ -81,26 +149,37 @@ const DQAContainer = ( ): React.Node => {
   );
 
   const setMetricVote = ( metric, vote ) => {
-    const qualityMetricParams = {
-      id: observationUUID,
-      metric,
-      agree: vote,
-      ttyl: -1
-    };
     setLoadingMetric( metric );
     if ( vote ) {
       setLoadingAgree( true );
     } else {
       setLoadingDisagree( true );
     }
+
+    // Special case for needs_id metric which is a fave/unfave kind of action
+    if ( metric === "needs_id" ) {
+      const faveParams = {
+        id: observationUUID,
+        scope: metric,
+        vote: vote === false
+          ? "no"
+          : "yes"
+      };
+      faveMutation.mutate( faveParams );
+      return;
+    }
+
+    const qualityMetricParams = {
+      id: observationUUID,
+      metric,
+      agree: vote,
+      ttyl: -1
+    };
```

Comment:
I don't object to the way you're doing this, but I think it might further confuse developers in this repo as to how this particular part of the DQA is different from all of the others. Maybe a better way to put it is that it doesn't help developers understand that the `needs_id` part of the DQA is different from all the other parts.